frontend/compositions/map/Map.tsx

```diff
@@ -0,0 +1,118 @@
+import useData from "./useData"
+import styles from "./map.module.css"
+
+export type BoundingType = [[number, number], [number, number]]
+export type D3CallableSelectionType = d3.Selection<Element, unknown, any, any>
+export type D3ZoomEventType = d3.D3ZoomEvent<Element, any>
+export type ZoomBehaviorType = d3.ZoomBehavior<Element, unknown>
+export type DispatchType = d3.Dispatch<d3.ZoomTransform>
+
+export default function Map() {
+  const data = useData()
+  const mapRef = useRef<HTMLDivElement>()
+  const { width, height } = useResizeObserver({ ref: mapRef })
+
+  const zoomRef = useRef<SVGSVGElement>()
+
+  const [markerParams, setMarkerParams] = useState<MarkerDescription[]>([])
+
+  const zoom = d3.zoom().on("zoom", zoomed)
+  const zoomContainer = d3.select("#zoom-container") as D3CallableSelectionType
+  zoomContainer.call(zoom)
+
+  function zoomed(event: D3ZoomEventType) {
+    const { transform } = event
+    zoomContainer.attr("transform", transform.toString())
+  }
+
+  const zoomResetButton = d3.select("#zoom-reset")
+  zoomResetButton.on("click", (event: MouseEvent) => {
+    zoomContainer.transition().duration(750).call(zoom.transform, d3.zoomIdentity)
+  })
+
+  const projection = d3
+    .geoAlbersUsa()
+    .scale(1300)
+    .translate([487.5 + 112, 305 + 50])
+
+  const path = d3.geoPath(projection)
+
+  useEffect(() => {
+    if (!data) return
+
+    const populationScale = d3
+      .scaleLinear()
+      .domain(d3.extent(data.features, (d) => d.properties.population))
+      .range([1, 25])
+
+    const getMarkerParamsFromFeature = (d: Feature) => {
+      const markerParams: MarkerDescription = {
+        geoCenter: projection((d.geometry as Point).coordinates as [number, number]),
+        dataPoint: populationScale(d.properties.population),
+        label: d.properties.city,
+      }
+      return markerParams
+    }
+
+    setMarkerParams(data.features.map((feature) => getMarkerParamsFromFeature(feature)))
+  }, [data])
+
+  const onMapClick = (event: MouseEvent) => {
+    const target = event.target as any
+    const d = target.__data__
+    if (target.classList.contains("state")) {
+      const [[x0, y0], [x1, y1]]: BoundingType = path.bounds(d)
+
+      event.stopPropagation()
+
+      zoomContainer
+        .transition()
+        .duration(750)
+        .call(
+          zoom.transform,
+          d3.zoomIdentity
+            .translate(width / 2, height / 2)
+            .scale(Math.min(8, 0.9 / Math.max((x1 - x0) / width, (y1 - y0) / height)))
+            .translate(-(x0 + x1) / 2, -(y0 + y1) / 2),
+          d3.pointer(event, zoomContainer.node())
+        )
+      data.filter.value = d.properties.name
+      data.setFilterProperties({ ...data.filter })
+    }
+  }
+
+  zoomContainer.on("click", onMapClick)
+
+  return (
+    <div
+      id="map-container"
+      className={styles.mapContainer}
+      ref={mapRef}
+      style={{ height: "80vh", width: "100vw" }}>
```

Comment:
Small nitpick: I know that mobile use-cases are not our primary concern, but we should avoid using `vh` in most cases since that value gets unpredictable on mobile devices: https://css-tricks.com/the-trick-to-viewport-units-on-mobile/

Reply:
Thanks for flagging this and sharing the article (I LOVE css-tricks).
We need a absolute size here, and we need to set it directly on the div (not in the CSS), so the solutions in this article wouldn't work in this case.  
Since it's not primary, and the task might be a pretty big one, we should hold on this and focus on primary features. The way the map fits into the rest of the layout will be changing the container anyway, so we should at least wait to see what we end up with then.

Reply:
Yeah, again not urgent, but something for us to keep in mind since it's such a major gotcha with vh